app/services/internal_proxy_proxy_selector.rb

```diff
@@ -0,0 +1,16 @@
+class InternalProxyProxySelector < ::RemoteExecutionProxySelector
+  def initialize
+    # TODO: Remove this once we have a reliable way of determining the internal proxy without katello
+    raise _('Internal proxy selector can only be used if Katello is enabled') unless defined?(::Katello)
+    super
+  end
+
+  def available_proxies(host, provider)
+    # TODO: Once we have a internal proxy marker/feature on the proxy, we can
```

Comment:
Is there a Foreman Redmine to add this we can reference here?

Reply:
Not yet, I first wanted to talk about if having such a marker is the way to go.